### ISSUE TITLE
Unknown default Storage Location on MacOS.

### ISSUE THREAD

**Issue by kkrawczyk123** [opendatakit/briefcase#275]
#### Software versions 
Briefcase v1.8, Java v1.8. operating system MacOS,

#### Problem description
Weird problem displayed to me, default storage location was different (probably non-existing folder)  than I've chosen before. I was able to change it and the only way to reproduce the problem for me was starting briefcase again.  

#### Steps to reproduce the problem
1. Create new folder
2. Clone there briefcase code from github
3. Build an app
4. Run the app and go to Settings

#### Expected behavior
Default storage location should equal to the last chosen one when run briefcase or window when you can choose it should display.

#### Other information 
Pull from aggregate succeed but I couldn't find where pulled forms are. Screenshot attached:
![mac](https://user-images.githubusercontent.com/33342488/35098016-da0deaf4-fc52-11e7-9aad-372b1924ad73.png)

**Comment by ggalmazor** [comment]
Hi @kkrawczyk123!

The problem you're experiencing is due to the automated test suite that we now have, which will overwrite your normal Briefcase settings whenever it's run.

I'm already working on the solution that will prevent the tests from overriding these settings.

**Comment by ggalmazor** [comment]
@opendatakit-bot claim

**Comment by ggalmazor** [comment]
Hi @kkrawczyk123!

Yesterday we got some PRs merged and you shouldn't have this problem anymore. Just be sure to remove your saved Java user preferences beforehand and don't forget to reboot before starting to test Briefcase. @yanokwa has some experience doing this on a mac and might provide help.

**Comment by yanokwa** [comment]
@kkrawczyk123 the files I removed are:
* ~/Library/Preferences/com.apple.java.util.prefs.plist
* ~/Library/Preferences/org.opendatakit.briefcase.plist

Can you remove these, reboot the machine, and see if you can still reproduce this issue?

**Comment by kkrawczyk123** [comment]
I tried to do the same but it seems that I couldn't find those files at all. Screenshot of contents of Library/Preferences folder on my mac attached. Any suggestions what I should do? 
![java_mac](https://user-images.githubusercontent.com/33342488/35212674-373eab04-ff5b-11e7-985d-488344d1c191.png)

I've also noticed very weird briefcase behavior when run an app. I couldn't choose any tab pull/push or export until I change storage location. I am not sure how to reproduce that because I just run briefcase and it did occur to me like that. The Video's quality is not the best but it shows the problem:
http://www.mediafire.com/file/71vz7j67kp6vx1l/VID_20180122_095217.mp4

**Comment by yanokwa** [comment]
@kkrawczyk123 There are two `/Library/Preferences/` folder in macOS. 

There is one at `/Library/Preferences/` and there is another at `~/Library/Preferences/` (note the tilde). The tilde means that you have to add the home folder. So in my case, the folder I need to look at is `/Users/yanokwa/Library/Preferences/`

My guess is that you are looking at `/Library/Preferences/` and not `~/Library/Preferences/`.

You can confirm those files exist by trying this at the command line.
* `ls ~/Library/Preferences/com.apple.java.util.prefs.plist`
* `ls ~/Library/Preferences/org.opendatakit.briefcase.plist`

You can delete those files exist by trying this at the command line.
* `rm ~/Library/Preferences/com.apple.java.util.prefs.plist`
* `rm ~/Library/Preferences/org.opendatakit.briefcase.plist`

The reason you can't select the export/import tabs in your video is because Briefcase can't write to that weird folder and it needs to write to a folder for pull/push/export. If you change the folder to something else, it should work.

**Comment by kkrawczyk123** [comment]
Thanks, @yanokwa, surely I was looking for those files in the wrong folder. I removed them via command line and reboot. Actually, I was able to reproduce that issue in 3 from 4 attempts. 
And I was able to open pull/push/export tabs no matter if I changed this unknown default storage location and pull and export processes were successful.
Additionally, I reproduced that issue on Ubuntu too. Screenshots attached. 
![screen shot 2018-01-23 at 10 03 03](https://user-images.githubusercontent.com/33342488/35268978-004b7976-002b-11e8-97bd-c88533befe3d.png)
![screen shot 2018-01-23 at 09 54 28](https://user-images.githubusercontent.com/33342488/35268991-06350abe-002b-11e8-80b4-4aae335c3b3a.png)
![storage location_ubuntu](https://user-images.githubusercontent.com/33342488/35269002-0b71a622-002b-11e8-9203-81197739c3da.png)

**Comment by yanokwa** [comment]
I am not able to reproduce this on a new install of macOS 10.13 that has never had Briefcase installed. To run the application, I did the following:
1. git clone https://github.com/opendatakit/briefcase.git (153-g3b29d73)
1. ./gradlew run
On launch, the Settings folder defaulted to `/Users/yanokwa/Desktop/ODK Briefcase Storage`. 

I then quit the app, and removed the preferences files
1. rm ~/Library/Preferences/com.apple.java.util.prefs.plist (it didn't exist)
1. rm ~/Library/Preferences/org.opendatakit.briefcase.plist (it existed)

After a reboot, when I launched, I got an error message about the storage folder missing and I was asked to choose a new one.

@grzesiek2010 @mmarciniak90 can you please help @kkrawczyk123 replicate this? Some things to try
1. Confirm the version you are using with `java --version`. Maybe there is some weird Java?
1. In addition to the prefs in the home folder (~), remove them in the root (`/Library/Preferences/com.apple.java.util.prefs.plist` and `/Library/Preferences/org.opendatakit.briefcase.plist`) as well and reboot.
1. Check out a fresh repo and do a `./gradlew clean run`.

**Comment by kkrawczyk123** [comment]
@yanokwa , @ggalmazor , I, @grzesiek2010  and @mmarciniak90 investigated that issue deeply. I reproduced steps that you described on mac and if fact it is the only way when briefcase actually asked for choosing storage location with the first run at all. I always use ./gradlew build and then java -jar commands to run briefcase so it is the only difference between our approaches. Now, when I tried to use ./gradlew build method on Mac, build failed so I didn’t even have a chance to see the default storage location. It froze with dialog window (screenshot attached). 
I was able to reproduce that on Ubuntu and Windows both. When I used ./gradlew run command on them I also saw this weird default storage location. I tried to run the build from circleCi on all systems and I also saw this weird storage location. Moreover, @grzesiek2010  and @mmarciniak90 reproduced that independently on their ubuntu’s too.
![mac_build](https://user-images.githubusercontent.com/33342488/35329430-1d0995fe-0100-11e8-84ac-6e5edabb9ccd.png)
